languages/cpp/exercises/concept/basics/basics.h

```diff
@@ -0,0 +1,14 @@
+#if !defined(BASICS_H_)
+#define BASICS_H_
```

Comment:
This will be syntax that is new to the student as I don't think it is introduced in the `introduction.md` file. I don't know much about C++, but is it possible to omit this extra syntax?

Reply:
Right, I completely forgot about include guards, took them for a granted. :)
By the way they're pretty important, they avoid the inclusion of the same code multiple times in the a translation unit.
If you omit them the below snippet would be "expand" 3 times instead of the first time only.

``` // basics.cpp
#include "basics.h"
#include "basics.h"
#include "basics.h"
```

It's a standard practice and think they should be at least mentioned in the introduction, without talking about macros though that would be a bit too much IMO.

Reply:
I could change it to use `#pragma once` instead, it does the same thing and is supported by [most compilers][supported-compilers] but it's not a recognized standard.

[supported-compilers]: https://en.wikipedia.org/wiki/Pragma_once#Portability

Reply:
I see two possibilities here:
1. You have the include already in the `basics.cpp`, maybe just leave the guard out altogether? Then a student learns not to add it _again_ because of all the errors it introduces. ... Or maybe this learning by error is not a good way for Exercism?
2. Leave the guards in and explain them later. If taking this route, maybe mention in the `introduction.md` that they are include guards and that they will be explained later.
I would recommend staying away from `#pragma once` as , as you mention, it is non-standard, and also it does not resolve the problem, there is still something additional in the code that the student needs to understand.

Reply:
I agree with the idea of explaining that header guard is something you have to have in a header file (with a reference to an explanation to why that is important).

On a subject of `#pragma once`, that is indeed not a standard and should not be taught as of this moment.

Reply:
I think the second option is the best, we mention it in passing and go more in depth in later exercises.